src/monopole_gr/monopole_gr.cpp

```diff
@@ -424,15 +491,50 @@ TaskStatus SpacetimeToDevice(StateDescriptor *pkg) {
         Real dbetadt = -0.5 * r * (alpha(i) * dKdt + K * dalphadt);
 
         // printf("%d: %.15e %.15e %.15e %.15e\n", i, dadt, dalphadt, dKdt, dbetadt);
-        gradients(Gradients::DADT, i) = force_static ? 0 : dadt;
-        gradients(Gradients::DALPHADT, i) = force_static ? 0 : dalphadt;
-        gradients(Gradients::DKDT, i) = force_static ? 0 : dKdt;
-        gradients(Gradients::DBETADT, i) = force_static ? 0 : dbetadt;
+        gradients(Gradients::DADT, i) = mask * dadt;
+        gradients(Gradients::DALPHADT, i) = mask * dalphadt;
+        gradients(Gradients::DKDT, i) = mask * dKdt;
+        gradients(Gradients::DBETADT, i) = mask * dbetadt;
       });
 
   return TaskStatus::complete;
 }
 
+TaskStatus CheckRateOfChange(StateDescriptor *pkg, Real dt) {
+  PARTHENON_REQUIRE_THROWS(pkg->label() == "monopole_gr",
+                           "Requires the monopole_gr package");
+  auto &params = pkg->AllParams();
+  auto enabled = params.Get<bool>("enable_monopole_gr");
+  if (!enabled) return TaskStatus::complete;
+
+  auto dtwarn = params.Get<bool>("dtwarn");
+  if (dtwarn && parthenon::Globals::my_rank == 0) {
+    auto alpha = params.Get<Alpha_t>("lapse");
+    auto alpha_last = params.Get<Alpha_t>("lapse_last");
+    auto npoints = params.Get<int>("npoints");
+    auto gradients = params.Get<Gradients_t>("gradients");
+    auto dtwarn_eps = params.Get<Real>("dtwarn_eps");
+
+    Real err;
+    parthenon::par_reduce(
+        parthenon::loop_pattern_flatrange_tag, "monopole_gr check evolution time scales",
+        parthenon::DevExecSpace(), 0, npoints - 1,
+        KOKKOS_LAMBDA(const int i, Real &lmax_err) {
+          Real diff_alpha = (alpha(i) - alpha_last(i)) / dt;
+          Real diff_dalpha = std::abs(diff_alpha - gradients(Gradients::DALPHADT, i));
```

Comment:
Should this be relative or absolute difference? Lapse could be small where the action is, right?

Reply:
Hmm. I'm not sure. You're right alpha could be small where the action is. I don't want to renormalize by dalpha/dt, because that would risk dividing by zero when things are well-behaved. But we could divide by alpha to try to capture that scale.

Reply:
I think your suggestion of absolute differences of logarithmic derivatives is good.